### PR TITLE
AvailableGames.java fixes/cleanups

### PR DESCRIPTION
**Pull request by DanVanAtta**
A set of fixes centered around a 'probable bug' static analysis warning in AvailableGames.java
Perhaps most easily reviewed commit by commit, but even in some perhaps not too bad.

Summary of updates:
- class variable rename
- two NPE guards
- remove an always true condition

### CODE REVIEW

File: src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java [hunk 1]
```diff
@@ -99,18 +99,23 @@ private static void populateAvailableGames(final Map<String, URI> availableGames
...
-    for (final File game : games.listFiles()) {
-      if (game.toURI() != null && game.isFile() && game.getName().toLowerCase().endsWith("xml")) {
-        final boolean added = addToAvailableGames(game.toURI(), availableGames, mapNamePropertyList);
-        if (added) {
-          availableMapFolderOrZipNames.add(mapDir.getName());
+    if (games.listFiles() != null) {
+      for (final File game : games.listFiles()) {
+        if (game.isFile() && game.getName().toLowerCase().endsWith("xml")) {
+          final boolean added = addToAvailableGames(game.toURI(), availableGames, mapNamePropertyList);
+          if (added) {
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java]: This line should be merged with the preceeding line...

**Comment by DanVanAtta** [comment]
It is possible for both methods to return null where null checks were added. Static analysis caught it. In one condition `games` would have to be a non-directory file for `listFiles()` to return null, which maybe or maybe won't ever be the case. Regardless, the problem is moot with the NPE check, will have same handling as empty return value.

File: src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java (hunk 1 above)
Comment (DanVanAtta) [inline src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java]: I think it is maybe debatable either way.  I'm partly inclined to update it, but at the same time this is just an indentation update here: https://github.com/triplea-game/triplea/pull/1972/files?w=1 

`addToAvailableGames` I think is a problematic method though, I think part of the reason it was on two lines was to convey that `addToAvailableGames` has a lot of side effects.

IMO not worth messing around with these two lines just yet unless we would do some deeper stuff.

File: src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java [hunk 2]
```diff
@@ -99,18 +99,23 @@ private static void populateAvailableGames(final Map<String, URI> availableGames
...
     if (!games.exists()) {
       // no games in this map dir
       return;
     }
-    for (final File game : games.listFiles()) {
-      if (game.toURI() != null && game.isFile() && game.getName().toLowerCase().endsWith("xml")) {
-        final boolean added = addToAvailableGames(game.toURI(), availableGames, mapNamePropertyList);
-        if (added) {
-          availableMapFolderOrZipNames.add(mapDir.getName());
+    if (games.listFiles() != null) {
```
Comment (ron-murhammer) [inline src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java]: Wouldn't it be better to check if games.isDirectory() and add that to the if statement right above?

Comment (ssoloff) [inline src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java]: It looks like if you don't extract the result of `games.listFiles()` to a local and use that in the if check and for-each loop, you have a TOCTOU.  Not likely to happen, but...

Note also that, per the [JDK](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#listFiles--), `File#listFiles()` can return `null` if an I/O error occurs, not just if the descriptor does not represent a directory.

Comment (DanVanAtta) [inline src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java]: > Note also that, per the JDK, File#listFiles() can return null if an I/O error occurs, not just if the descriptor does not represent a directory.

Given this, we need the null check anyways, the folder check would be redundant. TBH, this was a straight forward quick null check flagged by static analysis. I suspect we'll get merge conflicts soon. If that happens I don't know if I'll be willing to spend time to resolve them.

**Pull request merged**